server/src/db/implementation/mongo-db-extension.ts

```diff
@@ -15,21 +22,32 @@ export class MongoDbExtension implements DbExtension {
   public fooditemUnit: FoodItemUnitRepository;
   public socobouser: SocoboUserRepository;
   public recipe: RecipeRepository;
-  public recipeCategory: RecipeCategoryRepository;
+  public recipeCategory: CrudRepository<RecipeCategory>;
+  public recipeIngredient: CrudRepository<RecipeIngredient>;
 
   constructor (
     private _fooditemTemplateModel: Model<Document & FoodItemTemplate>,
     private _fooditemCategoryModel: Model<Document & FoodItemCategory>,
     private _fooditemUnitModel: Model<Document & FoodItemUnit>,
     private _socobouserModel: Model<Document & SocoboUser>,
     private _recipeModel: Model<Document & Recipe>,
-    private _recipeCategoryModel: Model<Document & RecipeCategory>
+    private _recipeCategoryModel: Model<Document & RecipeCategory>,
+    private _recipeIngredientModel: Model<Document & RecipeIngredient>
   ) {
     this.fooditemTemplate = new FoodItemTemplateRepository(_fooditemTemplateModel);
     this.fooditemCategory = new FoodItemCategoryRepository(_fooditemCategoryModel);
     this.fooditemUnit = new FoodItemUnitRepository(_fooditemUnitModel);
     this.socobouser = new SocoboUserRepository(_socobouserModel);
-    this.recipe = new RecipeRepository(_recipeModel),
-    this.recipeCategory = new RecipeCategoryRepository(_recipeCategoryModel);
+    this.recipe = new RecipeRepository(
```

Comment:
this looks a little strange. RecipeRepository needs the model and the CrudRepository, because the CrudRepository doesn't provide all methods needed inside the RecipeRepository. What if your CrudRepository provide the model as a public property? but this violates the law of demeter. what did you think?

Reply:
Yes that is a little bit annoying, but I think it's the most flexible way and it makes the dependencies very clear. I also thought about inheritance but if I maybe need to add a method to the curdRepository class which is only needed by some classes I would break the other classes. So I decided to use composition.

I think we only have these problems because we don't use any DI thing and need to construct our object be ourself. But I think that's currently not to bad.

Reply:
Yes, its not so Bad.